MC_data_new_TH3red.py

```diff
@@ -0,0 +1,662 @@
+                param_tree.Branch('theta',theta,"theta/F")
+                param_tree.Branch('phi',phi,"phi/F") 
+                param_tree.Branch('nhits_og',nhits_og,"nhits_og/I")
+                param_tree.Branch('xhits_og',xhits_og,"xhits_og[nhits_og]/F")
+                param_tree.Branch('yhits_og',yhits_og,"yhits_og[nhits_og]/F")
+                param_tree.Branch('zhits_og',zhits_og,"zhits_og[nhits_og]/F")
+                param_tree.Branch('EDepHit_og',EDepHit_og,"EDepHit_og[nhits_og]/F")            
+                
+                final_imgs=list();
+                
+                if opt.events==-1:
+                    totev=tree.GetEntries()
+                else:
+                    
+                    if opt.events<=tree.GetEntries():
+                        totev=opt.events
+                    else:
+                        totev=tree.GetEntries()
+                    
+
+    
+                for entry in range(0, totev): #RUNNING ON ENTRIES
+                    tree.GetEntry(entry)
+                    print(entry)
+                    eventnumber[0] = tree.eventnumber
+                    #FIXME
+                    particle_type[0] = 0
+                    energy_ini[0] = tree.ekin_particle[0]*1000
+                    phi_ini[0] = -999.
+                    theta_ini[0] = -999.
+                    phi_ini[0] = np.arctan2( (tree.y_hits[1]-tree.y_hits[0]),(tree.z_hits[1]-tree.z_hits[0]) )
+                    theta_ini[0] = np.arccos( (tree.x_hits[1]-tree.x_hits[0]) / np.sqrt( np.power((tree.x_hits[1]-tree.x_hits[0]),2) + np.power((tree.y_hits[1]-tree.y_hits[0]),2) + np.power((tree.z_hits[1]-tree.z_hits[0]),2)) )
+                    outtree.Fill()
+                    theta[0]=theta_ini[0]
+                    phi[0]=phi_ini[0]
+                    energy[0]=energy_ini[0]
+                    track_length_3D[0]=0.0
+                    proj_track_2D[0]=0.0
+                    nhits_og[0]=tree.numhits
+                    final_image=rt.TH2I('pic_run'+str(run_count)+'_ev'+str(entry), '', opt.x_pix, 0, opt.x_pix-1, opt.y_pix, 0, opt.y_pix-1) #smeared track with background
+
+                    x_hits_tr = tree.x_hits
+
+                    #print("Original Info on X")
+                    #print(x_hits_tr)
+
+                    if opt.randZ:
+                        opt.z_gem=0
+                        rand = 50 + random.random()*400
+                        X0= min(tree.x_hits)
+                        for ihit in range(0,tree.numhits):
+                            x_hits_tr[ihit]-=X0
+                            x_hits_tr[ihit]+=rand
+
+                    #print("Random Diffused")
+                    #print(x_hits_tr)
+                    #print(opt.z_gem)
+                    histname = "histo_cloud_pic_"+str(run_count)+"_ev"+str(int(entry))
+                    
+                                        
+                    yMinCl=1152+(min(tree.y_hits))*opt.y_pix/opt.y_dim-20
```

Comment:
In the master branch, we already implemented a way to apply the saturation effect to the smallest region containing all the electrons, by computing the max and min of the x,y,z coordinates **after the smearing effect is applied**. I think the way it is now implemented in the master is slightly better because there is no risk to lose electrons that (due to the smearing effect) arrive very far from the original position (before the smearing). @SamueleTorelli what do you think?